File: iosApp/iosApp/data/mappers/media/ThumbnailMapper.swift
//
//  ThumbnailMapper.swift
//  iosApp
//
//  Created by nVoxel on 25.04.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import TDLibKit
import TGDriveKit

class ThumbnailMapper {
    
    private let fileMapper: FileMapper
    
    init(fileMapper: FileMapper) {
        self.fileMapper = fileMapper
    }
    
    func toResponse(thumbnail: Thumbnail) -> CommonThumbnail? {
        return CommonThumbnail(
            width: Int32(thumbnail.width),
            height: Int32(thumbnail.height),
            file: fileMapper.toResponse(file: thumbnail.file)
        )
    }
}

File: source/data/mappers/src/jvmMain/kotlin/com/voxeldev/tgdrive/mappers/media/ThumbnailMapper.kt
package com.voxeldev.tgdrive.mappers.media

import org.drinkless.tdlib.TdApi.Thumbnail
import com.voxeldev.tgdrive.main.api.media.Thumbnail as CommonThumbnail

/**
 * @author nvoxel
 */
class ThumbnailMapper(
    private val fileMapper: FileMapper,
) {

    fun toResponse(thumbnail: Thumbnail): CommonThumbnail =
        CommonThumbnail(
            width = thumbnail.width,
            height = thumbnail.height,
            file = fileMapper.toResponse(file = thumbnail.file)
        )
}
